map/src/manager/SettingsManager.js

```diff
@@ -0,0 +1,213 @@
+        setChanges(changes);
+    }
+}
+
+export async function deleteFile(file, ctx, changes, setChanges) {
+    const response = await apiPost(`${process.env.REACT_APP_USER_API_SITE}/mapapi/delete-file`, '', {
+        params: {
+            name: file.name,
+            type: file.type,
+        },
+    });
+    if (response.status === 200) {
+        changes = changes.filter((f) => f.id !== file.id);
+        setChanges(changes);
+    }
+}
+
+export async function restoreFile({ file, ctx, changes, setChanges, fromTrash = false }) {
+    const response = await apiGet(`${process.env.REACT_APP_USER_API_SITE}/mapapi/restore-file`, {
+        params: {
+            name: file.name,
+            type: file.type,
+            updatetime: file.updatetimems,
+        },
+        dataOnErrors: true,
+    });
+    if (response.status === 200) {
+        const newVersion = {
+            file: response.data,
+            type: 'file',
+            isLast: false,
+            id: response.data.id,
+        };
+        if (fromTrash) {
+            changes = deleteVersionsFromMenu({ changes, id: file.id });
+            setChanges(changes);
+        } else {
+            const fileIndex = changes.findIndex((change) => change.type === 'file');
+            let newChanges = [];
+            if (fileIndex === -1) {
+                newChanges = [...changes, newVersion];
+            } else {
+                newChanges = [...changes.slice(0, fileIndex), newVersion, ...changes.slice(fileIndex)];
+            }
+            setChanges(newChanges);
+        }
+    } else {
+        ctx.setRoutingErrorMsg(response.data);
+    }
+}
+
+export async function emptyTrash({ ctx, changes, setChanges }) {
+    const files = changes
+        .filter((change) => change.file && change.type === 'file')
+        .map((change) => ({
+            name: change.file.name,
+            type: change.file.type,
+            updatetime: change.file.updatetimems,
+        }));
+    const response = await apiPost(`${process.env.REACT_APP_USER_API_SITE}/mapapi/empty-trash`, files, {
+        dataOnErrors: true,
+    });
+    if (response.status === 200) {
+        setChanges([]);
+    } else {
+        ctx.setRoutingErrorMsg(response.data);
+    }
+}
+
+function deleteVersionsFromMenu({ changes, name = null, id = null }) {
+    changes = changes.filter((f) => {
+        if (f.file) {
+            if (name) {
+                return f.file.name !== name;
+            }
+            if (id) {
+                return f.id !== id;
+            }
+        }
+        return true;
+    });
+    changes = changes.filter((f, index) => {
+        if (f.type === 'month') {
+            const nextFileIndex = changes.slice(index + 1).findIndex((f) => f.type === 'file');
+            return nextFileIndex !== -1;
+        }
+        return true;
+    });
+    return changes;
+}
+
+export function formatString(templateString, replacements) {
```

Comment:
Please add comment with examples of input/output to clarify what the magic format it is.